ebml/EbmlElement.h

```diff
@@ -220,6 +277,7 @@ class EBML_DLL_API EbmlCallbacks {
         inline EbmlElement & NewElement() const { return Create(); }
         /// is infinite/unknown size allowed
         inline bool CanHaveInfiniteSize() const { return CanInfinite; }
+        virtual const bool HasDefault() const { return false; }
```

Comment:
Compiler warning:

```
lib/libebml/ebml/EbmlElement.h:280:17: warning: 'const' type qualifier on return type has no effect [-Wignored-qualifiers]
virtual const bool HasDefault() const { return false; }
```

Just change it to `virtual bool HasDefault() const` both here & in `EbmlCallbacksDefault`.

Reply:
Additional compiler warning:

```
lib/libebml/ebml/EbmlElement.h:262:20: warning: 'libebml::EbmlCallbacks' has virtual functions but non-virtual destructor [-Wnon-virtual-dtor]
class EBML_DLL_API EbmlCallbacks {                                                                                                                                                                                                                                                                                              
```

Adding a simple

```
    virtual ~EbmlCallbacks() {}
```

seems to get rid of them, but at the cost of a compiler error in various later source files, e.g. `EbmlDummy.cpp`:

```
lib/libebml/src/EbmlDummy.cpp:13:1: fatal error: constexpr variable cannot have non-literal type 'const EbmlCallbacks'
DEFINE_EBML_CLASS_ORPHAN(EbmlDummy, 0xFF, 1, "DummyElement")
^
lib/libebml/ebml/EbmlElement.h:150:60: note: expanded from macro 'DEFINE_EBML_CLASS_ORPHAN'
#define DEFINE_EBML_CLASS_ORPHAN(x,id,idl,name)            DEFINE_xxx_CLASS_ORPHAN(x,id,idl,name,GetEbmlGlobal_Context)
                                                           ^
lib/libebml/ebml/EbmlElement.h:145:38: note: expanded from macro 'DEFINE_xxx_CLASS_ORPHAN'
    constexpr const EbmlCallbacks x::ClassInfos(x::Create, Id_##x, false, name, Context_##x); \
                                     ^
lib/libebml/ebml/EbmlElement.h:274:13: note: 'EbmlCallbacks' is not literal because it has a user-provided destructor
    virtual ~EbmlCallbacks() {}
            ^
```

Reply:
I fixed both. It's odd I don't get these warnings in clang.

Reply:
Hmmm the `warning: 'libebml::EbmlCallbacks' has virtual functions but non-virtual destructor [-Wnon-virtual-dtor]` warning is still there in revision 5b434b7a011de77864de2c14a14d880a71d9afed.

Reply:
Fixed. I also added #195 to make these issues more visible in the future.